src/pkg/zoci/copier.go

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package zoci contains functions for interacting with Zarf packages stored in OCI registries.
+package zoci
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/oci"
+	"github.com/defenseunicorns/zarf/src/pkg/utils/helpers"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// CopyPackage copies a zarf package from one OCI registry to another
+// TODO (@AustinAbro321) we should see if we should also copy the manifests for a complete copy in this function
```

Comment:
yes, `CopyPackage` should also copy the manifest from `src.root` --> `dst.root`

Reply:
`oci.Copy` is a generic mirroring / multithreaded copy operation. Since this is an opinionation on top of that, where we know we _always_ want to copy the manifests, we can do so

Reply:
much of the code at the top of `src/pkg/packager/publish.go` can now be simplified by bringing it into this func